cmd/kubeadm/app/cmd/config.go

```diff
@@ -571,11 +585,37 @@ type ImagesList struct {
 	cfg *kubeadmapi.InitConfiguration
 }
 
+// imageTextPrinter prints image info in a text form
+type imageTextPrinter struct {
+	output.TextPrinter
+}
+
+// PrintObj is an implementation of ResourcePrinter.PrintObj for plain text output
+func (itp *imageTextPrinter) PrintObj(obj runtime.Object, writer io.Writer) error {
+	_, err := fmt.Fprintf(writer, "%s:%s\n", obj.(*outputapiv1alpha1.Image).Name, obj.(*outputapiv1alpha1.Image).Version)
+	return err
+}
+
+// imageTextPrintFlags provides flags necessary for printing image in a text form.
+type imageTextPrintFlags struct{}
+
+// ToPrinter returns kubeadm printer for the text output format
+func (ipf *imageTextPrintFlags) ToPrinter(outputFormat string) (output.Printer, error) {
+	if outputFormat == output.TextOutput {
+		return &imageTextPrinter{}, nil
+	}
+	return nil, genericclioptions.NoCompatiblePrinterError{OutputFormat: &outputFormat, AllowedFormats: []string{output.TextOutput}}
+}
+
 // Run runs the images command and writes the result to the io.Writer passed in
-func (i *ImagesList) Run(out io.Writer) error {
+func (i *ImagesList) Run(out io.Writer, printer output.Printer) error {
 	imgs := images.GetControlPlaneImages(&i.cfg.ClusterConfiguration)
 	for _, img := range imgs {
-		fmt.Fprintln(out, img)
+		nameVer := strings.SplitN(img, ":", 2)
+		outputImage := outputapiv1alpha1.Image{Name: nameVer[0], Version: nameVer[1]}
```

Comment:
This is looking hacky to me. What if at some point images get pinned to digests (`image:v1.50.5` vs `image@sha256:ab82fd561234...`)?

Reply:
How you'd you suggest to parse this then?